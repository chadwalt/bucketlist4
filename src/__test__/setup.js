import { configure } from 'enzyme'
import Adapter from "enzyme-adapter-react-15"

configure({ adapter: new Adapter() })

global.window = {}
import localStorage from "mock-local-storage"
window.localStorage = global.localStorage