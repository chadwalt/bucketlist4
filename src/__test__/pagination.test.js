import React from "react"
import renderer from "react-test-renderer"
import Pagination from "../components/pagination"
import { StaticRouter } from "react-router"
import { shallow, mount } from "enzyme"
import "./setup"

describe("Login Snapshot", () => {
	it("renders correctly", () => {
		const props = {
			paginate: {
				prev_url: "",
				pages: 3,
				next_url: ""
			}
		}
		const paginate = renderer.create(<Pagination {...props} />)
		expect(paginate.toJSON()).toMatchSnapshot()
	})

	it("Should be able to click the prev button in pagination.", () => {
		const mockFunction = jest.fn()
		const props = {
			paginate: {
				prev_url: "/buckets/1",
				pages: 1,
				next_url: "/buckets/2"
			}
		}
		const element = mount(<Pagination {...props} onMovePage={mockFunction} />)
		element.find("#prev_page").simulate("click")
		element.find("#next_page").simulate("click")
		expect(mockFunction).toHaveBeenCalled()
	})

})