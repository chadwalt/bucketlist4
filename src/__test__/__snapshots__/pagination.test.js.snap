// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot renders correctly 1`] = `
<nav
  aria-label="Page navigation"
>
  <ul
    className="pagination"
  >
    
    <li
      className="page-item"
      id="page-1"
      onClick={[Function]}
    >
      <a
        className="page-link"
      >
        1
      </a>
    </li>
    
  </ul>
</nav>
`;
